qubesmanager/qube_manager.py

```diff
@@ -1246,16 +1246,44 @@ def action_shutdownvm_triggered(self):
             if reply == QMessageBox.Yes:
                 self.shutdown_vm(vm)
 
-    def shutdown_vm(self, vm, shutdown_time=vm_shutdown_timeout,
+    def get_connected_vms(self, vm, connected_vms):
+        for connected_vm in vm.connected_vms:
+            if connected_vm.is_running():
+                connected_vms.append(connected_vm)
+                self.get_connected_vms(connected_vm, connected_vms)
+
+    def shutdown_vm(self, vm, shutdown_time=vm_shutdown_timeout, force=False,
                     check_time=vm_restart_check_timeout, and_restart=False):
         try:
-            vm.shutdown()
+            connected_vms = []
+
+            if not and_restart:
+                self.get_connected_vms(vm, connected_vms)
+
+            if len(connected_vms) > 0:
+                reply = QMessageBox.question(
+                    self, self.tr("Qube Shutdown Confirmation"),
+                    self.tr("There are some qubes connected to <b>'{0}'</b>!"
+                        "<br><small>Do you want to shutdown: </small>"
+                        "<b>'{1}'</b>?").format(vm.name,
+                            ", ".join([x.name for x in connected_vms])),
+                    QMessageBox.Yes | QMessageBox.Cancel)
+
+                if reply != QMessageBox.Yes:
+                    return False
+
+                force = True
+                shutdown_time = shutdown_time * len(connected_vms)
```

Comment:
This looks a bit weird when shutting down all the VMs in parallel. If for example you have 10 of them, that shouldn't take 10x longer. The default shutdown timeout (60s) is already quite liberal, so maybe use just that (you are risking just a prompt to the user, not automatic kill), or max from all the involved VMs? If that really isn't enough, then maybe x2?

Reply:
uhM.. the default is 20s. I noticed the warning appearing with only 2 simultaneous shutdowns. Maybe we could multiply the total affected vms but set a max limit. maybe 1min?

Reply:
Oh, indeed qubes manager does not use `shutdown_timeout` VM property, but its own hardcoded value of 20s... Maybe change that?

Reply:
![lol](https://user-images.githubusercontent.com/35405566/109007812-d7afb080-76ac-11eb-807e-9e65cdec680d.png)

Is that enough? lol I don't know what happened :D

Reply:
I think qt uses ms not s for timers